application/src/main/java/org/togetherjava/tjbot/features/roleapplication/ApplicationCreateCommand.java

```diff
@@ -0,0 +1,297 @@
+            roleDisplayName = selectOption.getLabel();
+        } else {
+            roleDisplayName = "%s %s".formatted(emoji.getFormatted(), selectOption.getLabel());
+        }
+
+        Modal modal = Modal
+            .create(generateComponentId(event.getUser().getId(), roleDisplayName),
+                    String.format("Application form - %s", selectOption.getLabel()))
+            .addActionRow(ActionRow.of(body).getComponents())
+            .build();
+
+        event.replyModal(modal).queue();
+    }
+
+    /**
+     * Checks a given list of passed arguments (from a user) and calculates how many roles have
+     * missing data.
+     *
+     * @param args the list of passed arguments
+     * @return the amount of roles with missing data
+     */
+    private static long getIncorrectRoleArgsCount(final List<OptionMapping> args) {
+        final Map<String, Integer> frequencyMap = new HashMap<>();
+
+        args.stream()
+            .map(OptionMapping::getName)
+            .map(name -> name.split(VALUE_DELIMITER)[1])
+            .forEach(number -> frequencyMap.merge(number, 1, Integer::sum));
+
+        return frequencyMap.values().stream().filter(value -> value != 3).count();
+    }
+
+    /**
+     * Populates a {@link StringSelectMenu.Builder} with application roles.
+     *
+     * @param menuBuilder the menu builder to populate
+     * @param args the arguments which contain data about the roles
+     */
+    private void addRolesToMenu(StringSelectMenu.Builder menuBuilder,
+            final List<OptionMapping> args) {
+        final Map<Integer, MenuRole> roles = new HashMap<>();
+
+        for (int i = 0; i < args.size(); i += ARG_COUNT) {
+            OptionMapping optionTitle = args.get(i);
+            OptionMapping optionDescription = args.get(i + 1);
+            OptionMapping optionEmoji = args.get(i + 2);
+
+            roles.put(i,
+                    new MenuRole(optionTitle.getAsString(),
+                            generateComponentId(ROLE_COMPONENT_ID_HEADER,
+                                    optionTitle.getAsString()),
+                            optionDescription.getAsString(),
+                            Emoji.fromFormatted(optionEmoji.getAsString())));
+        }
+
+        roles.values()
+            .forEach(role -> menuBuilder.addOption(role.title(), role.value(), role.description(),
+                    role.emoji()));
+    }
+
+    private boolean handleHasPermissions(SlashCommandInteractionEvent event) {
+        Member member = event.getMember();
+        Guild guild = event.getGuild();
+
+        if (member == null || guild == null) {
+            return false;
+        }
+
+        if (!member.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("You do not have the required manage role permission to use this command")
+                .setEphemeral(true)
+                .queue();
+            return false;
+        }
+
+        Member selfMember = guild.getSelfMember();
```

Comment:
I don't think selfMember is correct, it should be event.member() or something. Self member is the bot itself. Double check.